src/msg/async/AsyncMessenger.cc

```diff
@@ -494,6 +504,16 @@ void AsyncMessenger::_finish_bind(const entity_addrvec_t& bind_addrs,
   entity_addrvec_t newaddrs = *my_addrs;
   for (auto& a : newaddrs.v) {
     a.set_nonce(nonce);
+    if (public_addrs) {
+      // transplantate network layer addresses while keeping ports
+      // (as they can be figured out by msgr from the allowed range)
```

Comment:
I agree that keeping the port is a good idea but I'm not sure what "as they can be figured out by msgr from the allowed range" means.  Can you elaborate?

Reply:
Added some details to the comment:

```cpp
      // transplantate network layer addresses while keeping ports
      // (as they can be figured out by msgr from the allowed range [1])
      // unless they are explicitly specified (NATing both IP/port?)
      //
      // [1]: the low-level `Processor::bind` scans for free ports in
      // a range controlled by ms_bind_port_min and ms_bind_port_max)
```